kbs_protocol/src/lib.rs

```diff
@@ -177,6 +181,18 @@ impl KbsRequest for KbsProtocolWrapper {
 
         bail!("Request KBS: Attested but KBS still return Unauthorized")
     }
+
+    async fn attest(&mut self, url_string: String) -> Result<String> {
+        let url = Url::parse(&url_string)
```

Comment:
why do we need to parse the url and not pass it directly to `self.attestation`? I don't know what you imagine the caller of this function would do.